src/bt-proto.c

```diff
@@ -436,3 +436,23 @@ append_bt_uuid_t(struct pdu* pdu, const bt_uuid_t* uuid)
 
   return append_to_pdu(pdu, "m", uuid->uu, sizeof(uuid->uu));
 }
+
+long
+append_bt_ifname(struct pdu* pdu, const char* ifname)
+{
+  static const uint8_t padding[IFNAME_LEN] = {0};
+  int len, padding_len;
+
+  len = strnlen(ifname, IFNAME_LEN);
+  if (len < 0) {
+    ALOGD("strnlen < 0");
+    return -1;
+  }
+  else if (len < IFNAME_LEN) {
+    padding_len = IFNAME_LEN - len;
+    return append_to_pdu(pdu, "mm", ifname, (size_t)len, padding, (size_t)padding_len);
+  }
+
+  ALOGD("Interface name is more than 17 byte");
```

Comment:
This should be 'ALOGE'. The error messages in this file start with non-capital letters. '17' should be read from the constant.